project_cost_report/static/src/js/report_action.js

```diff
@@ -0,0 +1,386 @@
+        var button = $(QWeb.render("projectCostReport.printButton", {}));
+        button.bind('click', () => this.downloadPDF());
+        return button;
+    },
+    /**
+     * Render the FOLD button.
+     *
+     * returns {jQuery} the button node
+     */
+    renderFoldButton(){
+        var button = $(QWeb.render("projectCostReport.foldButton", {}));
+        button.bind('click', () => this.fold());
+        return button;
+    },
+    /**
+     * Render the UNFOLD button.
+     *
+     * returns {jQuery} the button node
+     */
+    renderUnfoldButton(){
+        var button = $(QWeb.render("projectCostReport.unfoldButton", {}));
+        button.bind('click', () => this.unfold());
+        return button;
+    },
+    /**
+     * Render the buttons of the control panel.
+     *
+     * returns {Array<jQuery>} an array of button nodes
+     */
+    getControlPanelButtons(){
+        if(!this.controlPanelButtons){
+            this.printButton = this.renderPrintButton();
+            this.foldButton = this.renderFoldButton();
+            this.unfoldButton = this.renderUnfoldButton();
+            this.controlPanelButtons = [this.printButton, this.foldButton, this.unfoldButton];
+        }
+        return this.controlPanelButtons;
+    },
+    /**
+     * Update the control pannel of the report view.
+     */
+    updateControlPanel(){
+        this.update_control_panel({
+            breadcrumbs: this.getParent().get_breadcrumbs(),
+            cp_content: {$buttons: this.getControlPanelButtons()},
+        });
+    },
+    /**
+     * Hide/show the FOLD/UNFOLD buttons given the folding state.
+     *
+     * The folding state is managed with the attribute `unfolded`.
+     * This method must be called after setting the attribute
+     * in order to update the DOM nodes.
+     */
+    updateFoldButtonVisibility(){
+        if(this.unfolded){
+            this.foldButton.show();
+            this.foldButton.css('display', 'inline-block');
+            this.unfoldButton.hide();
+        }
+        else {
+            this.foldButton.hide();
+            this.unfoldButton.show();
+            this.unfoldButton.css('display', 'inline-block');
+        }
+    },
+    /**
+     * Start the widget.
+     *
+     * This is a standard method of web.Widget.
+     * It is called when the widget is ready to be rendered.
+     */
+    start(){
+        var result = this._super();
+        this.updateControlPanel();
+        this.updateHtml();
+        this.updateFoldButtonVisibility();
+        return result;
+    },
+    /**
+     * Refresh the report.
+     */
+    async refresh(){
+        this.updateHtml();
+        this.updateFoldButtonVisibility();
+    },
+    /**
+     * Download the PDF version of the report.
+     */
+    downloadPDF(){
+        framework.blockUI();
+        session.get_file({
+            url: "/web/project_cost_report/" + this.projectId,
+            data: {report_context: JSON.stringify(this.reportContext)},
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'report_context' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3675949)